Exercise 4.6 Part 2 Step 3

In [7]:
# Import libraries

import pandas as pd
import numpy as np
import os

In [15]:
# Step 3
# importing new frames
# create path
path =  r'C:\Users\Marker\Videos\Instakart Basket Analysis 01-25'
# Import dataset products_checked.csv
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_checked.csv'), index_col = False)
# Import dataset orders_products_combined.pkl
df_merge_large = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))

In [21]:
# Import dataset products_checked.csv
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_checked.csv'), index_col = False)

In [17]:
df_merge_large.head()

Unnamed: 0  order_id  user_id eval_set  order_number  order_dow  \
0           0   2539329        1    prior             1          2   
1           0   2539329        1    prior             1          2   
2           0   2539329        1    prior             1          2   
3           0   2539329        1    prior             1          2   
4           0   2539329        1    prior             1          2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     7.0         196                  1   
1                  8                     7.0       14084                  2   
2                  8                     7.0       12427                  3   
3                  8                     7.0       26088                  4   
4                  8                     7.0       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

In [31]:
#Step 4

In [19]:
df_merge_large.shape

(32434489, 12)

In [ ]:
#Step 5

In [23]:
df_prods.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [25]:
# Drop unnecessary columns
df_prods = df_prods.drop(['Unnamed: 0'], axis=1)

In [27]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [29]:
df_prods.shape

(49672, 5)

In [37]:
# Drop unnecessary columns
df_merge_large = df_merge_large.drop(['_merge'], axis=1)

In [39]:
df_merge_large.head()

Unnamed: 0  order_id  user_id eval_set  order_number  order_dow  \
0           0   2539329        1    prior             1          2   
1           0   2539329        1    prior             1          2   
2           0   2539329        1    prior             1          2   
3           0   2539329        1    prior             1          2   
4           0   2539329        1    prior             1          2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     7.0         196                  1   
1                  8                     7.0       14084                  2   
2                  8                     7.0       12427                  3   
3                  8                     7.0       26088                  4   
4                  8                     7.0       26405                  5   

   reordered  
0          0  
1          0  
2          0  
3          0  
4          0

In [43]:
# Drop unnecessary columns
df_merge_large = df_merge_large.drop(['Unnamed: 0'], axis=1)

In [45]:
df_merge_large.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2539329        1    prior             1          2                  8   
2   2539329        1    prior             1          2                  8   
3   2539329        1    prior             1          2                  8   
4   2539329        1    prior             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered  
0                     7.0         196                  1          0  
1                     7.0       14084                  2          0  
2                     7.0       12427                  3          0  
3                     7.0       26088                  4          0  
4                     7.0       26405                  5          0

In [47]:
# Drop unnecessary columns
df_merge_large = df_merge_large.drop(['eval_set'], axis=1)

In [49]:
df_merge_large.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2539329        1             1          2                  8   
2   2539329        1             1          2                  8   
3   2539329        1             1          2                  8   
4   2539329        1             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered  
0                     7.0         196                  1          0  
1                     7.0       14084                  2          0  
2                     7.0       12427                  3          0  
3                     7.0       26088                  4          0  
4                     7.0       26405                  5          0

In [51]:
df_merge_large.shape

(32434489, 9)

In [53]:
# Merge the updated dataframes
df_ords_prods_merge = df_prods.merge(df_merge_large, on = 'product_id', indicator = True)

In [54]:
# Check the output
df_ords_prods_merge.head()

product_id                product_name  aisle_id  department_id  prices  \
0           1  Chocolate Sandwich Cookies        61             19     5.8   
1           1  Chocolate Sandwich Cookies        61             19     5.8   
2           1  Chocolate Sandwich Cookies        61             19     5.8   
3           1  Chocolate Sandwich Cookies        61             19     5.8   
4           1  Chocolate Sandwich Cookies        61             19     5.8   

   order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   3139998      138            28          6                 11   
1   1977647      138            30          6                 17   
2    389851      709             2          0                 21   
3    652770      764             1          3                 13   
4   1813452      764             3          4                 17   

   days_since_prior_order  add_to_cart_order  reordered _merge  
0                     3.0                  5          0   both  
1                    20.0                  1          1   both  
2                     6.0                 20          0   both  
3                     7.0                 10          0   both  
4                     9.0                 11          1   both

In [58]:

df_ords_prods_merge.shape

(32404859, 14)

In [60]:
#Step 6

In [62]:
df_ords_prods_merge['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

The resulting dataframe has 32,404,859 rows, and each of those rows have information found in both input data sets, as we used an inner join for this merge.

In [73]:
#Step 7
# Export data to .pkl
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'ords_prods_merge.pkl'))